### ISSUE TITLE
Various issues with Config as Code API

### ISSUE THREAD

**Issue by ldhertert** [ldhertert/harness-cli#7]
Can't get read content for certain paths (i.e. `harness config:get --path 'Setup/Cloud Providers/cv-demo.yaml'`)

API Auth is not working for some parts of the API, such as fetching config as code tree for applications. (https://harness.atlassian.net/browse/DX-2211)

**Comment by ldhertert** [comment]
I've created a new ticket specifically to track progress of supporting API Key authentication for all YAML endpoints: https://harness.atlassian.net/browse/PL-13358

**Comment by ldhertert** [comment]
I have closed https://harness.atlassian.net/browse/DX-2218, as I'm now using a different API that can read all content.  It is still blocked by the API key auth issue.
